From a0e74d2129a721f017ff7c0d19d35e896ec0cee6 Mon Sep 17 00:00:00 2001
From: Nishanth Menon <nm@ti.com>
Date: Tue, 4 Nov 2014 11:47:17 -0600
Subject: [PATCH 1297/1587] ARM: OMAP2+: DRA7XX: OPP: Add encoding for "All
 OPP" configuration

Value 23 indicates all OPPs. Add that to the configuration.

Based on Vayu Speed grade document revision 0.2 (May 30, 2014).

Signed-off-by: Nishanth Menon <nm@ti.com>
Acked-by: Dave Gerlach <d-gerlach@ti.com>
[zou: Original patch taken from
ti-sdk-am335x-evm-08.00.00.00-Linux-x86-Install.bin]
Signed-off-by: Cao Zou <cao.zou@windriver.com>
---
 arch/arm/mach-omap2/opp7xx_data.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/opp7xx_data.c b/arch/arm/mach-omap2/opp7xx_data.c
index 42bf3ff..f65be76 100644
--- a/arch/arm/mach-omap2/opp7xx_data.c
+++ b/arch/arm/mach-omap2/opp7xx_data.c
@@ -32,6 +32,7 @@
  */
 #define EFUSE_HAS_OD_MPU_OPP		11
 #define EFUSE_HAS_HIGH_MPU_OPP		15
+#define EFUSE_HAS_ALL_MPU_OPP		23
 
 /*
  * We only include OPP_OD and OPP_HIGH here, OPP_NOM is common to all
@@ -74,6 +75,7 @@ int __init dra7xx_opp_init(void)
 		max_freq >>= SPEED_VAL_SHFT;
 
 		switch (max_freq) {
+		case EFUSE_HAS_ALL_MPU_OPP:
 		case EFUSE_HAS_HIGH_MPU_OPP:
 			opp_def_list_enable_opp(dra7xx_es1_0_opp_list,
 				ARRAY_SIZE(dra7xx_es1_0_opp_list),
-- 
1.7.5.4

